modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/RotateGestureRecognizer.java

```diff
@@ -307,10 +298,11 @@ else if (initialInertiaRotationVelocity < -MAX_INITIAL_VELOCITY)
                         totalRotation += currentRotation;
                         sendRotateEvent(false);
                         angleReference = newAngle;
-                        double timePassed = ((double)time - rotationStartTime) / 1000000000;
-                        if (timePassed > 1e-4) {
-                            initialInertiaRotationVelocity = currentRotation / timePassed;
-                            rotationStartTime = time;
+                        long nanosPassed = nanos - rotationStartNanos;
+
+                        if (nanosPassed > INITIAL_VELOCITY_THRESHOLD_NANOS) {
+                            initialInertiaRotationVelocity = currentRotation / nanosPassed * NANOS_TO_SECONDS;
```

Comment:
is this correct?  shouldn't it be `currentRotation / (nanosPassed * NANOS_TO_SECONDS)` ?